clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -578,21 +587,25 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
 
     if (memref.exit.type == TRACE_TYPE_THREAD_EXIT) {
         shard->saw_thread_exit_ = true;
-        report_if_false(shard,
-                        !TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
-                                 shard->file_type_) ||
-                            shard->found_instr_count_marker_,
-                        "Missing instr count markers");
-        report_if_false(shard,
-                        shard->found_cache_line_size_marker_ ||
-                            (shard->skipped_instrs_ && !is_a_unit_test(shard) &&
-                             shard->stream->get_cache_line_size() > 0),
-                        "Missing cache line marker");
-        report_if_false(shard,
-                        shard->found_page_size_marker_ ||
-                            (shard->skipped_instrs_ && !is_a_unit_test(shard) &&
-                             shard->stream->get_page_size() > 0),
-                        "Missing page size marker");
+        if (!TESTANY(OFFLINE_FILE_TYPE_KERNEL_SYSCALL_TRACE_TEMPLATES,
+                     shard->file_type_)) {
+            report_if_false(
+                shard,
+                !TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
+                         shard->file_type_) ||
+                    shard->found_instr_count_marker_,
+                "Missing instr count markers");
+            report_if_false(shard,
+                            shard->found_cache_line_size_marker_ ||
+                                (shard->skipped_instrs_ && !is_a_unit_test(shard) &&
+                                 shard->stream->get_cache_line_size() > 0),
+                            "Missing cache line marker");
```

Comment:
Hmm, why doesn't this fire on the context switch templates too?  The unit test doesn't have them.  Oh actually the real thing does:

```
$ bin64/drrun -t drcachesim -simulator_type view -infile ../src/clients/drcachesim/tests/mock_switch_sequences.x64.zip
Output format:
<--record#-> <--instr#->: <---tid---> <record details>
------------------------------------------------------------
           1           0:     3198712 <marker: version 6>
           2           0:     3198712 <marker: filetype 0xe40>
           3           0:     3198712 <marker: cache line size 64>
           4           0:     3198712 <marker: chunk instruction count 10000000>
           5           0:     3198712 <marker: page size 4096>
...
```

Maybe the syscall ones should follow suit?  At least add a comment about doing that to avoid needing an exception here, since the switch templates don't need this exception.

Reply:
I modified syscall templates to have page size and cache line size.